### PR TITLE
ValidCount assert and TempFile creation without file creation

### PR DESCRIPTION
**Pull request by Lexey**

### CODE REVIEW

File: Main/src/IO/TempData.cs [hunk 1]
```diff
@@ -134,6 +134,8 @@ public sealed class TempFile : TempBase
 		{
 			private FileInfo _info;
 
+			/// <summary>Create an instance using an automatically constructed temp file path</summary>
```
Comment (ig-sinicyn) [inline Main/src/IO/TempData.cs]: Dot at the end of summary :)

File: Main/src/IO/TempData.cs [hunk 2]
```diff
@@ -171,15 +173,19 @@ protected override void DisposePath(string path, bool disposing)
 		#endregion
 
 		#region Factory methods
-		private static string GetTempName() => Guid.NewGuid() + ".tmp";
+		/// <summary>Creates a temporary file or directory name</summary>
+		/// <returns>The temporary name</returns>
+		/// <remarks>The resulting name is a local name (does not include a base path)</remarks>
+		public static string GetTempName() => Guid.NewGuid() + ".tmp";
```
Comment (ig-sinicyn) [inline Main/src/IO/TempData.cs]: Should this be public at all? If yes, the `<summary/>` should be updated.

File: Main/src/Assertions/Code.cs [hunk 3]
```diff
@@ -170,6 +170,25 @@ public static partial class Code
 		}
 		#endregion
 
+		#region Argument validation - valid count
+		/// <summary>Asserts if the passed value is not a valid count</summary>
```
Comment (ig-sinicyn) [inline Main/src/Assertions/Code.cs]: There should be a dot ('.') at the end of `<summary/>` content. This way documentation looks better.

File: Main/src/Assertions/Code.cs [hunk 4]
```diff
@@ -170,6 +170,25 @@ public static partial class Code
...
+		public static void ValidCount(int count, [NotNull, InvokerParameterName] string argName)
+			=> InRange(count, argName, 0, int.MaxValue);
+
+		/// <summary>Asserts if the passed value is not a valid count</summary>
+		/// <param name="count">The count value.</param>
+		/// <param name="argName">The name of the argument.</param>
+		/// <param name="length">The length.</param>
+		[DebuggerHidden, MethodImpl(AggressiveInlining)]
+		[AssertionMethod]
+		public static void ValidCount(int count, [NotNull, InvokerParameterName] string argName,
```
Comment (ig-sinicyn) [inline Main/src/Assertions/Code.cs]: Line wrapping, something like this, may be?
```
		public static void ValidCount(
			int count,
			[NotNull, InvokerParameterName] string argName,  
			int length) => 
				InRange(count, argName, 0, length);  
```

**Comment by ig-sinicyn** [comment]
There're some minor fixes todo, after that - 👍

File: Main/src/IO/TempData.cs (hunk 2 above)
Comment (Lexey) [inline Main/src/IO/TempData.cs]: It is useful to construct temporary files in custom locations. So, it is useful to make it public.

**Comment by ig-sinicyn** [comment]
LGTM. Waiting for approval from @andrewvk.

**Pull request merged**